packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -417,65 +417,148 @@ contract OPContractsManager is ISemver {
-            // Starting with the permissioned game, permissioned weth, and anchor state registry, which all chains have.
+            // Note that, the code below uses several independently scoped blocks to avoid stack too deep errors.
+
+            // All chains have the Permissioned Dispute Game. We get it first so that we can use it to
+            // retrieve its WETH and the Anchor State Registry when we need them.
             IPermissionedDisputeGame permissionedDisputeGame = IPermissionedDisputeGame(
                 address(
                     getGameImplementation(
                         IDisputeGameFactory(opChainAddrs.disputeGameFactory), GameTypes.PERMISSIONED_CANNON
                     )
                 )
             );
-            IDelayedWETH delayedWETHPermissionedGameProxy = permissionedDisputeGame.weth();
-            IAnchorStateRegistry anchorStateRegistryProxy = permissionedDisputeGame.anchorStateRegistry();
-            upgradeTo(proxyAdmin, address(anchorStateRegistryProxy), impls.anchorStateRegistryImpl);
-            upgradeTo(proxyAdmin, address(delayedWETHPermissionedGameProxy), impls.delayedWETHImpl);
-            // TODO: redeploy and replace permissioned game implementation
+            // We're also going to need the l2ChainId below, so we cache it in the outer scope.
+            uint256 l2ChainId = permissionedDisputeGame.l2ChainId();
+
+            // Replace the Anchor State Registry Proxy with a new Proxy and Implementation
+            // For this upgrade, we are replacing the previous Anchor State Registry, thus we:
+            // 1. deploy a new Anchor State Registry implementation
+            // 2. get the starting anchor root corresponding to the currently respected game type.
+            // 3. initialize the proxy with that anchor root
+            IAnchorStateRegistry newAnchorStateRegistryProxy;
+            {
+                // Deploy a new proxy, because we're replacing the old one.
+                newAnchorStateRegistryProxy = IAnchorStateRegistry(
+                    deployProxy({
+                        _l2ChainId: permissionedDisputeGame.l2ChainId(),
+                        _proxyAdmin: _opChains[i].proxyAdmin,
+                        _saltMixer: "v2.0.0",
+                        _contractName: "AnchorStateRegistry"
+                    })
+                );
+
+                // Get the starting anchor root by:
+                // 1. getting the anchor state registry from the Permissioned Dispute Game.
+                // 2. getting the respected game type from the OptimismPortal.
+                // 3. getting the anchor root for the respected game type from the Anchor State Registry.
+                {
+                    GameType gameType = IOptimismPortal2(payable(opChainAddrs.optimismPortal)).respectedGameType();
+                    (Hash root, uint256 l2BlockNumber) = permissionedDisputeGame.anchorStateRegistry().anchors(gameType);
```

Comment:
nit, we make `.anchorStateRegistry()` calls in a few places, could create a helper method to dedupe if that helps with code size